src/server/game/Entities/Player/Player.cpp

```diff
@@ -6478,6 +6478,7 @@ void Player::CheckAreaExploreAndOutdoor()
                     XP = uint32(sObjectMgr->GetBaseXP(areaEntry->area_level)*sWorld->getRate(RATE_XP_EXPLORE));
                 }
 
+                XP = uint32(XP * (IsEventActive(sWorld->getIntConfig(CONFIG_RATE_XP_WEEKEND_EVID)) ? sWorld->getRate(RATE_XP_WEEKEND) : 1.0f));
```

Comment:
Alright, but make it more easy to later expand as well. Get rid of the compact syntax, and expand it into an if branch.

```
uint32 getFinalizedXP(uint32 initialXP)
{
    if (whatever)
        initialXP *= whatever;
    return initialXP;
```